content/news/20.4networking-rework.md

```diff
@@ -0,0 +1,224 @@
+```java
+@SubscribeEvent
+public static void register(final RegisterPacketHandlerEvent event) {
+    final IPayloadRegistrar registrar = event.registrar("my_mod")
+            .versioned("1.2.3")
+            .optional();
+}
+```
+{{< notice note >}}
+The registrar is a semi-immutable object; calling `versioned(String version)` or `optional()` on an instance of the registrar will cause a new instance with the desired configuration to be created.
+{{< /notice >}}
+
+{{< notice warning >}}
+The registrar loses its validity the moment the scope of the event has been left. Registering payload handlers outside the event handling scope will result in those payloads not being known to the system and not being sent over the connection. Additionally, the attempt to register them outside the event scope will trigger an exception.
+{{< /notice >}}
+
+The registrar offers six different endpoints, three pairs of 2, to register new payloads. A pair exists for the play phase, one pair for the special configuration sub-phase, and one pair of methods for both.
+
+{{< notice info >}}
+It is impossible to register custom payloads that should be sent during the login phase of the connection. And the new code offers, as such, no infrastructure to achieve this.
+{{< /notice >}}
+
+Within each pair of registration methods for a phase, two variants (as such, six methods) are available. One that registers the same handler for both sides of the connection and one that takes a consumer, allowing for the configuration of single-sided or differentially handled payloads.
+
+An example of the signature of the methods for the configuration phase is as follows:
+```java
+<T extends CustomPacketPayload> IPayloadRegistrar configuration(ResourceLocation id, FriendlyByteBuf.Reader<T> reader, IConfigurationPayloadHandler<T> handler);
+
+<T extends CustomPacketPayload> IPayloadRegistrar configuration(ResourceLocation id, FriendlyByteBuf.Reader<T> reader, Consumer<IDirectionAwarePayloadHandlerBuilder<T, IConfigurationPayloadHandler<T>>> handler);
+```
+For the play phase, similar methods exist.
+For payloads that are supposed to be sent during both the play phase and the configuration phase, there also exists a pair.
+However, here, the handler is a common supertype of the handling callbacks, and the handler only has a reduced common superset of the information available to both phase handlers to it available.
+
+### Payload discrimination
+When considering this system, you might, rightfully so, ask how the system keeps the different payload types apart from each other. 
+This is achieved by the system itself. 
+It writes a discriminator ID to the connection before invoking the write method for a payload. 
+Conversely on the client side, the discriminator is read first, before a reader is looked up so that the rest of the payload can be read.
```

Comment:
`Conversely` implies two opposite systems, yet here they're symmetric.
```suggestion
Similarly, on the client side, the discriminator is read first, then a reader is looked up so that the rest of the payload can be read.
```